httpclient/httpclient_test.go

```diff
@@ -276,9 +275,10 @@ func TestClient_Call_Retry500(t *testing.T) {
 		BaseURL: server.URL,
 		Timeout: time.Second,
 	})
-	req := NewRequest("POST", "/", time.Millisecond)
+	req := client.NewRequest("POST", "/")
+	req.Timeout = time.Millisecond
 	ctx := testcontext.Background()
-	err := client.Call(ctx, req)
+	err := req.Call(ctx)
```

Comment:
I'm not sure I like this change. I'd rather behavioural functions all lived on the client, while value oriented functions lived on the request

Reply:
is that a problem in practice? I personally feel like this API is slightly easier to use